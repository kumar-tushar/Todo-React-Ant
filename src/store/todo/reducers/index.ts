import { combineReducers } from 'redux';
import { todoReducer } from './todo.reducer';

export const rootReducer = combineReducers({
  todo: todoReducer
});

export type RootState = ReturnType<typeof rootReducer>;
