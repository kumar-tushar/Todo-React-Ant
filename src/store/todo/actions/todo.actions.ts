import { ITodo } from '../models/todo.model';
import { TodoActionTypes } from './todo.actions.types';
import { TODO_ACTION_TYPES } from '../constants';

export function addTodo(todo: ITodo): TodoActionTypes {
  return {
    type: TODO_ACTION_TYPES.ADD_TODO_SUCCESS,
    payload: todo,
  };
}

export function removeTodo(todo: ITodo): TodoActionTypes {
  return {
    type: TODO_ACTION_TYPES.REMOVE_TODO_SUCCESS,
    payload: todo,
  };
}

export function toggleTodoStatus(todo: ITodo): TodoActionTypes {
  return {
    type: TODO_ACTION_TYPES.TOGGLE_TODO_STATUS_SUCCESS,
    payload: todo,
  };
}
