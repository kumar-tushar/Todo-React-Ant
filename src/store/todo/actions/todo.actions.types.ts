import { TODO_ACTION_TYPES } from 'store/todo/constants';
import { ITodo } from 'store/todo/models/todo.model';

interface IAddTodoAction {
  type: typeof TODO_ACTION_TYPES.ADD_TODO_SUCCESS;
  payload: ITodo;
}

interface IToggleTodoAction {
  type: typeof TODO_ACTION_TYPES.TOGGLE_TODO_STATUS_SUCCESS;
  payload: ITodo;
}

interface IRemoveTodoAction {
  type: typeof TODO_ACTION_TYPES.REMOVE_TODO_SUCCESS;
  payload: ITodo;
}

export type TodoActionTypes =
  | IAddTodoAction
  | IRemoveTodoAction
  | IToggleTodoAction;
