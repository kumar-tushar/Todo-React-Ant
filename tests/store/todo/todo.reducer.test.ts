import { removeTodo, toggleTodoStatus } from '../../../src/store/todo/actions';
import {
  initialState,
  todoReducer,
} from '../../../src/store/todo/reducers/todo.reducer';

import { createTestTodo } from './factories/todo.factory';

describe('todo reducer', () => {
  it('should handle addTodo', () => {
    expect(
      todoReducer(
        initialState,
        createTestTodo({
          id: '440351',
          name: 'This is a test todo',
          completed: true,
        })
      )
    ).toMatchSnapshot();
  });

  it('should handle removeTodo', () => {
    const createdTodoAction = createTestTodo({});

    let todoReducerState = todoReducer(initialState, createdTodoAction);

    expect(todoReducerState).toBeDefined();
    expect(todoReducerState.todos).toHaveLength(1);

    todoReducerState = todoReducer(
      initialState,
      removeTodo(createdTodoAction.payload)
    );

    expect(todoReducerState.todos).toHaveLength(0);
  });

  it('should handle toggleTodoStatus', () => {
    const createTodoAction = createTestTodo({
      id: '1234',
    });

    const state = todoReducer(initialState, createTodoAction);

    const todoReducerState = todoReducer(
      state,
      toggleTodoStatus(createTodoAction.payload)
    );

    expect(todoReducerState.todos[0].completed).toBe(
      !createTodoAction.payload.completed
    );
  });
});
