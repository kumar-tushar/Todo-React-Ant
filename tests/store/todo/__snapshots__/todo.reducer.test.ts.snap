// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo reducer should handle addTodo 1`] = `
Object {
  "todos": Array [
    Object {
      "completed": true,
      "id": "440351",
      "name": "This is a test todo",
    },
  ],
}
`;
